File: Python/multiRL/_modify_colnames.py
"""Internal column-name conversion helpers."""


def _modify_colnames(data, colnames):
    if colnames is None:
        colnames = {}

    out = {
        "subid": "Subject",
        "block": "Block",
        "trial": "Trial",
        "object": None,
        "reward": None,
        "action": "Action",
        "exinfo": None,
    }
    out.update(colnames)

    if out["object"] is None:
        out["object"] = [
            name for name in list(data.columns)
            if str(name).startswith("Object_")
        ]

    if out["reward"] is None:
        out["reward"] = [
            name for name in list(data.columns)
            if str(name).startswith("Reward_")
        ]

    return out

File: R/R/modify_colnames.R
.modify_colnames <- function(data, colnames) {
  default_colnames <- list(
    subid = "Subject",
    block = "Block",
    trial = "Trial",
    object = NA_character_,
    reward = NA_character_,
    action = "Action",
    exinfo = NA_character_
  )
  out <- utils::modifyList(default_colnames, colnames)

  if (base::length(out$object) == 1L && base::is.na(out$object)) {
    out$object <- base::grep(
      "^Object_",
      base::names(data),
      value = TRUE
    )
  }

  if (base::length(out$reward) == 1L && base::is.na(out$reward)) {
    out$reward <- base::grep(
      "^Reward_",
      base::names(data),
      value = TRUE
    )
  }

  out
}

.modify_data_id <- function(data, id, subid) {
  if (!base::is.null(id)) {
    data <- data[data[[subid]] %in% id, , drop = FALSE]
  }

  if (base::nrow(data) == 0L) {
    base::stop("No data rows remain after id filtering.")
  }

  data
}
